packages/shorebird_cli/lib/src/aab/aab_differ.dart

```diff
@@ -0,0 +1,105 @@
+import 'dart:convert';
+import 'dart:io';
+
+import 'package:archive/archive_io.dart';
+import 'package:path/path.dart' as p;
+import 'package:shorebird_cli/src/aab/mf_reader.dart';
+
+/// Types of code changes that we care about.
+enum AabDifferences {
+  none,
+  dart,
+  native,
+  assets,
+}
+
+/// Finds differences between two AABs.
+///
+/// Types of changes we care about:
+///   - Dart code changes
+///      - libapp.so will be different
+///   - Java/Kotlin code changes
+///      - .dex files will be different
+///   - Assets
+///      - **/assets/** will be different
+///      - AssetManifest.json will have changed if assets have been added or
+///        removed
+///
+/// Changes we don't care about:
+///   - Anything in META-INF
+///   - BUNDLE-METADATA/com.android.tools.build.libraries/dependencies.pb
+///      - This seems to change with every build, regardless of whether any code
+///        or assets were changed.
+///
+/// See https://developer.android.com/guide/app-bundle/app-bundle-format for
+/// reference.
+class AabDiffer {
+  /// Match files that change when Dart code changes.
+  final _dartChangePatters = [
+    RegExp(r'.+libapp\.so$'),
+    RegExp(r'.+libflutter\.so$'),
```

Comment:
Wouldn't the basename always be `libapp.so` and `libflutter.so`?

Reply:
Yep. Even thought this would be pretty trivial to handle without a RegExp, I opted to use them in all of the matchers since we have some cases (and possible more that aren't handled here) that are not as straightforward to detect.

Reply:
but is this regexp correct? Wouldn't this match `mylibapp.so` for example?

Reply:
Ahhhh I see what you mean. I can make this a bit more sophisticated.

Reply:
Fixed, lmk what you think